plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExBudgetingDefaults.hs

```diff
@@ -19,3 +26,9 @@ defaultCostModelParams = extractModelParams defaultCostModel
 -- import           Data.Default
 -- defaultCostModel :: CostModel
 -- defaultCostModel = def
+
+-- | Default costs for CEK machine instructions.
+defaultCekCosts :: CekCosts
```

Comment:
The `CostModel`/`CekCosts` thing is confusing. It could be `BuiltinCostModel` and `CekStepCostModel`, combined together to make a `CekCostModel` or something?

Reply:
I think we should definitely rename `CostModel` to `BuiltinCostModel`.  However if we do that there's a type parameter called `costs` that appears in a lot of place which we should logically rename `builtincosts`, and that might be a bit unwieldy.  Maybe we could call it `bic` instead...

I'm not so sure about combining the two things.  Internally they serve quite different purposes: the CEK costs are only used in the CEK machine and the builtin costs are used by the constant application machinery, which definitely doesn't need to know how much CEK operations are going to cost.  I considered making a single JSON file that contained both things, but that would complicate the parsing.  Also, if we had some other evaluator then it'd (probably) use the same model for builtins but a different one for machine steps, so it seemed safer to keep the two things apart.  It might make sense to read the two things in separately and combine them into a pair so that stuff outside the machine only needs to deal with one object which gets taken apart inside `Cek.hs` or somewhere.

Reply:
I've now renamed `CostModel` to `BultinCostModel` (and similalarly for a number of related functions) and renamed `CekCosts` to `CekMachineCosts`, but I haven't attempted to combine the two.

Reply:
From the point of view of the ledger, there's only one set of cost model parameters. Now, if we were to do Plutus V2 that has the same builtin application machinery but a different machine, then in principle they could reuse the model for the builtins, but in practice they'll just have a completely different set of parameters for that. So I do think we need a "big blob of everything" for the ledger, even if we keep them completely separate internally.